File: leetcode/two_sum/javascript/solve.js
const twoSum = function(nums, target) {
    let i = 0;
    while (i < nums.length) {
        for (let j = i + 1; j < nums.length; j++) {
            if (nums[i] + nums[j] === target) return [i, j];
        }
        i++;
    }
};

console.log(twoSum([2,7,11,15], 9));
console.log(twoSum([3,2,4], 6));
console.log(twoSum([3,3], 6));

File: leetcode/two_sum/go/solve.go
package main

import ("fmt")


func twoSum(nums []int, target int) []int {
	result := []int{}
	for i:= 0; i < len(nums); i++ {
		for j := i + 1; j < len(nums); j++ {
			if nums[i] + nums[j] == target {
				result = append(result, i, j)
			}
		}
	}
	return result
}


func main() {
	testOneNums := []int {2,7,11,15}
	testOneTarget := 9

	testTwoNums := []int {3,2,4}
	testTwoTarget := 6

	testThreeNums := []int {3, 3}
	testThreeTarget := 6

	fmt.Println(twoSum(testOneNums, testOneTarget));
	fmt.Println(twoSum(testTwoNums, testTwoTarget));
	fmt.Println(twoSum(testThreeNums, testThreeTarget));
}